redbox-core/redbox/models/file.py

```diff
@@ -22,12 +22,21 @@ class ChunkMetadata(BaseModel):
     """
 
     uuid: UUID = Field(default_factory=uuid4)
-    index: int
-    file_name: str
-    page_number: int | None = None
+    index: int = 0  # The order of this chunk in the original resource
     created_datetime: datetime.datetime = datetime.datetime.now(datetime.UTC)
-    token_count: int
     chunk_resolution: ChunkResolution = ChunkResolution.normal
+    creator_type: str
+    original_resource_ref: str  # URL or file name
+    token_count: int
+
+
+class UploadedFileMetadata(ChunkMetadata):
+    """
+    Model for uploaded document chunk metadata.
+    """
+
+    page_number: int | None = None
     name: str
     description: str
     keywords: list[str]
+    creator_type: str = "uploaded_file"
```

Comment:
```suggestion
    creator_type: Literal["Wikpedia", "USER UPLOADED DOCUMENT", "gov.uk"] = "USER UPLOADED DOCUMENT"
```

to match https://github.com/i-dot-ai/redbox/blob/40f0914e25b2e6d473151e563dfe6625b0512bed/django_app/redbox_app/redbox_core/models.py#L735